packages/platform-browser/lazy-animations/src/async_animation_renderer.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import type {ɵAnimationRendererFactory as AnimationRendererFactory, ɵcreateEngine as createEngine, ɵAnimationRenderer as AnimationRenderer} from '@angular/animations/browser';
+import {Injectable, NgZone, Renderer2, RendererFactory2, RendererStyleFlags2, RendererType2} from '@angular/core';
+
+type FromAnimationBrowserModule = {
+  ɵcreateEngine: typeof createEngine,
+  ɵAnimationRendererFactory: typeof AnimationRendererFactory
+};
+
+@Injectable()
+export class AsyncAnimationRendererFactory implements RendererFactory2 {
+  private _moduleLoaded: Promise<AnimationRendererFactory>|null = null;
+
+  constructor(
+      private doc: Document, private delegate: RendererFactory2, private zone: NgZone,
+      private animationType: 'animations'|'noop',
+      private moduleImpl?: Promise<FromAnimationBrowserModule>) {}
+
+  /**
+   * @internal
+   */
+  private loadImpl(
+      type: 'animations'|'noop',
+      moduleImpl: Promise<FromAnimationBrowserModule> = import('@angular/animations/browser')):
+      Promise<AnimationRendererFactory> {
+    return moduleImpl
+        .catch((e) => {
+          // TODO: Create a runtime error
+          throw new Error('Failed to load the @angular/animations/browser module');
+        })
+        .then(({ɵcreateEngine, ɵAnimationRendererFactory}) => {
+          // We can't create the renderer yet because we might need the hostElement and the type
+          // Both are provided in createRenderer().
+          const engine = ɵcreateEngine(type, this.doc);
+          const rendererFactory = new ɵAnimationRendererFactory(this.delegate, engine, this.zone);
+          this.delegate = rendererFactory;
+          return rendererFactory;
+        });
+  }
+
+  /**
+   * This method is delegating the renderer creation to the factories.
+   * It uses default factory while the animation factory isn't loaded
+   * and will rely on the animation factory once it is loaded.
+   *
+   * Calling this method will trigger as side effect the loading of the animation module
+   * if the renderered component uses animations.
+   */
+  createRenderer(hostElement: any, rendererType: RendererType2): Renderer2 {
+    const renderer = this.delegate.createRenderer(hostElement, rendererType);
+
+    if ((renderer as AnimationRenderer).isAnimationRenderer) {
+      // The factory is already loaded, this is an animation renderer
+      return renderer;
+    }
+
+    // We need to prevent the DomRenderer to throw an error because of synthetic properties
+    if ('throwOnSyntheticProps' in renderer) {
```

Comment:
I'd propose this refactoring to make the code more compatible with Closure Compiler optimizations that rename fields:
```suggestion
    if (typeof renderer.throwOnSyntheticProps === 'boolean') {
```